internal/manifest/crypto_test.go

```diff
@@ -96,3 +98,43 @@ func newKey(t *testing.T, bits int) *rsa.PrivateKey {
 	t.Logf("generated key: %q", HexString(x509.MarshalPKCS1PrivateKey(key)))
 	return key
 }
+
+func TestSeedShareKeyParseMarshal(t *testing.T) {
+	keyData, err := NewSeedShareOwnerPrivateKey()
+	require.NoError(t, err)
+
+	key, err := ParseSeedshareOwnerPrivateKey(keyData)
+	require.NoError(t, err)
+
+	pubHexStr := MarshalSeedShareOwnerKey(&key.PublicKey)
+
+	_, err = ParseSeedShareOwnerKey(pubHexStr)
+	require.NoError(t, err)
+
+	publicKeyHexStr, err := ExtractSeedshareOwnerPublicKey(keyData)
+	require.NoError(t, err)
+
+	publicKeyBytes, err := publicKeyHexStr.Bytes()
+	require.NoError(t, err)
+	publicKeyPem := pem.EncodeToMemory(&pem.Block{Type: "PUBLIC KEY", Bytes: publicKeyBytes})
+	_, err = ExtractSeedshareOwnerPublicKey(publicKeyPem)
```

Comment:
Nit: we could also test that the extracted public key is the same in both cases.